src/main/java/org/testng/xml/XmlWeaver.java

```diff
@@ -0,0 +1,80 @@
+package org.testng.xml;
+
+import org.testng.TestNGException;
+import org.testng.internal.ClassHelper;
+import org.testng.internal.reflect.ReflectionRecipes;
+
+/**
+ * A Utility class that helps represent a {@link XmlSuite} and {@link XmlTest} as String.
+ */
+final class XmlWeaver {
+    private static IWeaveXml instance = null;
```

Comment:
Why not instanciate it directly?

Reply:
The current implementation resorts to lazy model. So unless and until `toXml()` is invoked, none of this is going to be executed.

Reply:
Ok. For my information, why do you choose a lazy model? For performance reasons?

Reply:
Yes I chose it for performance reasons. I thought we can resort to figuring out the serialization logic only when there's a use for it.